src/python/bam_to_raw_fastq.py

```diff
@@ -46,39 +46,38 @@ def write_read(fastq, read):
                      'sequence': read.query_sequence})
         
     fastq.write('@{name}\n{sequence}\n+\n{quality}\n'.format(**info))
-    
-def check_putative_barcode(barcode_str, barcode_matching_dict):
+
+def check_putative_barcode(barcode_str, barcode_exact_dict, barcode_mismatch_dict):
     """
-    Procedure: check exact match of barcode, then 1 mismatch, then 1bp left/right shift
+    Procedure: check exact match of barcode, then 1 mismatch, then 1bp left/right
+     shift
     """
-    # check exact location first
     exact = True
-    corrected_barcode = barcode_matching_dict.get(barcode_str[1:9]) 
-    if corrected_barcode is None:
+    value = barcode_exact_dict.get(barcode_str[1:9]) # check exact location first
+    if value is None:
         exact = False
-        # check 1bp shift left
-        corrected_barcode = barcode_matching_dict.get(barcode_str[:8]) 
-        if corrected_barcode is None:
-            exact = False
-            # check 1bp shift right
-            corrected_barcode = barcode_matching_dict.get(barcode_str[2:])
-    return corrected_barcode, exact
+        value = barcode_mismatch_dict.get(barcode_str[1:9]) # check mismatch
+        if value is None:
+            value = barcode_exact_dict.get(barcode_str[:8]) # check 1bp shift left
+            if value is None:
+                value = barcode_exact_dict.get(barcode_str[2:]) # check 1bp shift right
+    return value, exact
 
-def create_barcode_matching_dict(barcode_list):
+def create_barcode_dicts(barcode_list):
```

Comment:
This function is used here and in correct fastq. That means that we need to always make sure that if we change one we will change also the other.  The best thing would be to put them in a util file and source it here.